doc/boot-partition.md

```diff
@@ -130,21 +134,34 @@ i.e. valid for all architectures (s390, see below)
 - order: /boot, swap, /
 - ext4
 
-x86-legacy
-----------
-- grub boot.img (stage 1) in MBR
-- install core.img (stage 1.5) in free space between partition table and 1. partition
-- if there isn't enough free space -> create additional partition
-- install all grub parts on same disk
-
-x86-efi
--------
-if system efi partition is already there -> reuse it
-if not create:
-- /boot/efi
-- fat32
-- size: 33 MB, 200MB
-- limit: below 2TB
+## x86-legacy
+
+*[this assumes grub will be installed into mbr]*
+
+- required size for grub embedding (currently): 84 kB; to be safe, check for >= 256 kB
+- if dos partition table
+    - embed in unused space before 1st partition
+    - fail if not enough space
```

Comment:
Here I'd like to add that grub2-install would not fail immediately but will seek other possibilities. In most filesystems that the blocklist will be used to locate core.img on that filesystem. If the filesystem cannot support blocklist, for eg the btrfs and zfs, then it will really fail. For that we also carried a patch in grub2-install for btrfs specifically for this issue, it will fallback to use the bootloader pad to embed the image if the mbr gap is too small and that's even better fallback than blocklist imho. :)